runtime/interpreter/value.go

```diff
@@ -14186,8 +14234,14 @@ func (v *CompositeValue) GetMember(interpreter *Interpreter, locationRange Locat
 
 	function, ok := v.Functions[name]
 	if ok {
+		var base *EphemeralReferenceValue
 		var self MemberAccessibleValue = v
-		return NewBoundFunctionValue(interpreter, function, &self)
+		if v.Kind == common.CompositeKindAttachment {
+			base = v.getBaseValue(interpreter, locationRange)
+			// in attachment functions, self is a reference value
+			self = NewEphemeralReferenceValue(interpreter, false, v, interpreter.MustSemaTypeOfValue(v))
+		}
```

Comment:
Maybe refactor into a function, and also use above https://github.com/onflow/cadence/pull/2113/files#diff-424575c293dc2bc3a28aab845829d7656fa83a7be4e026de8906935e23bbd52eR14100-R14103